spec/latest/json-ld-framing/index.html

```diff
@@ -552,34 +543,109 @@
+    <ol class="algorithm">
+      <li>If <em>property</em> is <code>@type</code>:
+        <ol class="algorithm">
+          <li><em>Property</em> matches if the <code>@type</code> property in frame includes any <a>IRI</a> in <em>values</em>.</li>
+          <li>Otherwise, <em>property</em> matches if <em>values</em> is not empty and the <code>@type</code> property in <em>frame</em> is <code><a>wildcard</a></code>.</li>
+          <li>Otherwise, <em>property</em> matches if <em>values</em> is empty and the <code>@type</code> property in <em>frame</em> is <code><a>match none</a></code>.</li>
+          <li>Otherwise, <em>property</em> does not match.</li>
+        </ol>
+      </li>
+      <li>Otherwise, the value of <em>property</em> in <em>frame</em> MUST be empty, or an array
+        containing a valid <a>frame</a>.</li>
+      <li><em>Property</em> matches if <em>values</em> is empty, or non existant,
+        the value of <em>property</em> in <em>frame</em>
+        is a <a>dictionary</a> containing only the key <code>@default</code> with any value,
+        and any other property in <em>node</em> has a non-default match.</li>
+      <li><em>Node</em> does not match if <em>values</em> is not empty and the value of <em>property</em> in <em>frame</em> is <code><a>match none</a></code>, and further matching is aborted.</li>
+      <li>Otherwise, <em>property</em> matches if <em>values</em> is not empty and the value of <em>property</em> in <em>frame</em> is <code><a>wildcard</a></code>.</li>
+      <li>Otherwise, if the value of <em>property</em> in <em>frame</em> is a <a>value pattern</a> (<em>value pattern</em>):
+        property matching is determined using the <a href="#value-matching">Value matching algorithm</a>. </li>
+      <li>Otherwise, for any <strong>node pattern</strong> (<em>node pattern</em>) which is one of the values of <em>property</em> in <em>frame</em>:
+        <ol class="algorithm">
+          <li>Let <em>value subjects</em> be the list of subjects from the <a>map of flattened subjects</a> matching the <a>node object</a> values from <em>values</em>.</li>
+          <li>Let <em>matched subjects</em> be the result of calling this algorithm recursively using
+            <em>state</em>, <em>value subjects</em> for <em>subjects</em>, <em>node pattern</em> for <em>frame</em>, and the <em>requireAll</em> flag.</li>
+          <li><em><em>Property</em></em> matches if <em>matched subjects</em> is not empty.</li>
+        </ol>
+      </li>
+      <li>Otherwise, <em>property</em> does not match.</li>
+    </ol>
+  </li>
+</ol>
+</section>
+
+<section id="value-matching">
+  <h2>Value Pattern Matching Algorithm</h2>
+
+  <p>The Value Pattern Matching Algorithm is used as part of the <a href="#framing-algorithm">Framing</a>
+    and <a href="#frame-matching">Frame Matching</a> algorithms. A value object
+    matches a value pattern using the <code><a>match none</a></code>, <code><a>wildcard</a></code>
```

Comment:
There are only two patterns listed here, but a comma is used. Is there supposed to be a third?